src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp

```diff
@@ -712,7 +712,7 @@ class npc_unworthy_initiate : public CreatureScript
                             }
                         }
 
-                        if (prison)
+                        if (prison && prison->getLootState() != GO_NOT_READY)
```

Comment:
```suggestion
                        if (prison && prison->getLootState() == GO_ACTIVATED)
```

The [ResetDoorOrButton](https://github.com/azerothcore/azerothcore-wotlk/blob/7712c2a97346a69ff1e2bca731d3f7e2aa09c494/src/server/game/Entities/GameObject/GameObject.cpp#L1404) works only when the object is GO_ACTIVATED, maybe we can check that directly

Reply:
> The [ResetDoorOrButton](https://github.com/azerothcore/azerothcore-wotlk/blob/7712c2a97346a69ff1e2bca731d3f7e2aa09c494/src/server/game/Entities/GameObject/GameObject.cpp#L1404) works only when the object is GO_ACTIVATED, maybe we can check that directly

Not quite, when a gameobject is first created (but before the first update) loot state is set to GO_NOT_READY which is what triggers this issue.

I do believe ResetDoorOrButton function in general probably should check for GO_NOT_READY (or by just checking if state != GO_ACTIVATED like you mentioned) but who knows what that could potentially break.